staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5348,6 +5348,57 @@ type NodeList struct {
 	Items []Node `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
 
+// +k8s:conversion-gen:explicit-from=net/url.Values
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeLogViewerOptions is the query options for a Node's logs REST call. The options specified here intersect unless
+// called out otherwise.
+type NodeLogViewerOptions struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// SinceTime is an RFC3339 timestamp from which to show logs.
+	// +optional
+	SinceTime *metav1.Time `json:"sinceTime,omitempty" protobuf:"bytes,1,opt,name=sinceTime"`
+
+	// UntilTime is an RFC3339 timestamp until which to show logs.
+	// +optional
+	UntilTime *metav1.Time `json:"untilTime,omitempty" protobuf:"bytes,2,opt,name=untilTime"`
+
+	// TailLines is used to retrieve the specified number of lines (not more than 100k) from the end of the log.
+	// Support for this is implementation specific and only available for Linux and Windows service logs. It will be
```

Comment:
I have only mentioned Linux and Windows because that is what I have implemented it for. Without adding to the implementation for other operating systems, I doubt it will work. My view was if someone else added support for other OSes, they would update the comment.